include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -4722,6 +4722,24 @@ struct OrtApi {
    * \param[in] adapter OrtLoraAdapter instance
    */
   ORT_API2_STATUS(RunOptionsAddActiveLoraAdapter, _Inout_ OrtRunOptions* options, _In_ const OrtLoraAdapter* adapter);
+
+  /// @}
+  /// \name OrtEpDynamicOptions
+  /// @{
+
+  /** \brief Set DynamicOptions for EPs
+   *
+   * \param[in] session
+   * \param[in] list of keys represented by null-terminated strings
+   * \param[in] list of values represented by null-terminated strings
+   * \param[in] number of key-value pairs
+   *
+   * \since Version xxx
+   * @TODO: update version number
+   * @TODO: should it be SetExecutionProvider... instead of SetEp...?
+   */
+  ORT_API2_STATUS(SetEpDynamicOptions, _Inout_ OrtSession* sess, _In_reads_(kv_len) const char* const* keys,
+                  _In_reads_(kv_len) const char* const* values, _In_ size_t kv_len);
```

Comment:
How does a user discover valid list of keys and values? Are you going to add a separate file?

Reply:
The file 'include\onnxruntime\core\session\onnxruntime_session_options_config_keys.h' has a description